calcom/cal.com
# fix: fetch event type with a team permissions

## What does this PR do?

This fixes #12416. There are 2 issues with the permissions when getting an EventType with a team field. First, the `req.query` includes an `id` field that is the ID of the event type is getting fetched yet the zod schema uses `.strict()` and errors with this unexpected extra field. This is the error outlined in the issue. Second, the endpoint is only checking that a membership object of role `MEMBER` exists rather than `MEMBER` or `ADMIN` or `OWNER`. Anyone in the team should be able to read so it makes sense to include all the roles when checking for membership.

Fixes #12416 

No UI update.

## Requirement/Documentation

Bug fix, no documentation update needed.

## Type of change


## How should this be tested?

I tested locally by using the seed database and user ID 12 who is owner of team 1. Login as user `teampro3@example.com` (userId 12) and create an API key. POST to `http://localhost:3002/v1/event-types` with body

```
{
  "length": 15,
  "title": "From postman",
  "slug": "rr-slug",
  "schedulingType": "ROUND_ROBIN",
  "teamId": 1,
  "hosts": [
    {
      "userId": 12,
      "isFixed": true
    }
  ]
}
```

to create a new EventType that is tied to a team. Note the returned ID for the new event type. Do a GET to `http://localhost:3002/v1/event-types/<YOUR_NEW_ID>` and notice that this fails on main but is fixed in this branch.

I cannot find any tests for this method but I would be happy to update if they exist and you can point me to them.

## Mandatory Tasks
